server/nodejs/resources/js/survey-edit.js

```diff
@@ -0,0 +1,381 @@
+function renderSurvey(surveyData) {
+    const formTitle = document.getElementById('formTitle');
+    const questionsContainer = document.getElementById('questionsContainer');
+
+    // Clear existing questions
+    questionsContainer.innerHTML = '';
+
+    // Set the title from the JSON data
+    formTitle.textContent = surveyData.survey_title;
+
+    // Iterate over the questions and add them to the DOM
+    surveyData.questions.forEach((questionData) => {
+        addQuestionFromData(questionData, questionsContainer);
+    });
+}
+
+// Function to initialize the survey rendering process
+function initializeSurvey() {
+    const questionsContainer = document.getElementById('questionsContainer');
+
+    // Try to fetch survey data from the server/API first
+    fetchSurveyData().then(data => {
+        if (data && data.survey_title) {
+            // If data is available from the server, render it
+            renderSurvey(data);
+        } else {
+            // If no data from server, use fallback local JSON data
+            console.log('Using fallback data from local JSON');
+            renderSurvey(surveyData);
+        }
+    });
+}
+
+// Function to create a new question container
+function createQuestionContainer(questionData) {
+    const newQuestionContainer = document.createElement("div");
+    newQuestionContainer.classList.add("question-container");
+    newQuestionContainer.setAttribute("data-question-id", questionData.question_id || questionIndex);
+
+    newQuestionContainer.innerHTML = `
+        <div class="question-content">
+            <div class="question-header">
+                <input type="text" class="question-text" value="${questionData.question_text}" onchange="updateQuestionText(this)">
+                <select class="question-type" onchange="changeQuestionType(this)">
+                    <option value="multiple_choice" ${questionData.question_type === 'multiple_choice' ? 'selected' : ''}>Multiple Choice</option>
+                    <option value="essay" ${questionData.question_type === 'essay' ? 'selected' : ''}>Essay</option>
+                    <option value="rating" ${questionData.question_type === 'rating' ? 'selected' : ''}>Rating</option>
+                </select>
+            </div>
+            <div class="options"></div>
+            <div class="button-container" style="display: ${questionData.question_type === 'multiple_choice' ? 'block' : 'none'};">
+                <button class="option-button" onclick="addOption(this)">Add Option</button>
+                <span>or</span>
+                <button class="option-button" onclick="addOtherOption(this)">Add Other</button>
+            </div>
+        </div>
+        <div class="side-buttons">
+            <button class="side-button add-question-btn" onclick="addNewQuestion(this)">+</button>
+            <button class="side-button remove-question-btn" onclick="removeQuestion(this)">x</button>
+        </div>
+    `;
```

Comment:
@loudiman  what are your thoughts on this, it seems like it needs to be decomposed further or something, this part of this code is hard to read